var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -53,24 +55,26 @@ class Samrai(Package):
     depends_on("zlib")
     depends_on("hdf5+mpi")
     depends_on("boost")
+    depends_on("m4")
```

Comment:
What does "type" do for me? This depends on m4 ` "--with-M4=%s" % spec['m4'].prefix,`.